deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c

```diff
@@ -13,59 +24,118 @@ static inline void _EmptyIterator
 	iter->nnz_idx = 0 ;
 }
 
-GrB_Info GxB_MatrixTupleIter_init
+static inline GrB_Info _MatrixTupleIter_init
 (
 	_GxB_MatrixTupleIter *iter,      // iterator to init
-	GrB_Matrix A                    // matrix to iterate over
+	GrB_Matrix A,                    // matrix to iterate over
+	int sparsity_type                // The sparsity type of the matrix
 ) {
-	GB_WHERE(A, "GxB_MatrixTupleIter_init (iter, A)") ;
+	GB_WHERE(A, "_MatrixTupleIter_init (iter, A)") ;
 	GrB_Index nrows ;
 	GrB_Matrix_nrows(&nrows, A) ;
 	GrB_Matrix_nvals(&iter->nvals, A) ;
 
-	GxB_Matrix_Option_get(A, GxB_SPARSITY_CONTROL, &iter->sparsity_type) ;
-	if(iter->sparsity_type != GxB_SPARSE && iter->sparsity_type != GxB_HYPERSPARSE) {
-		GB_ERROR (GrB_INVALID_VALUE,
-			"Invalid sparsity type: %d", iter->sparsity_type) ;
-	}
+	iter->sparsity_type = sparsity_type;
 	iter->A = A ;
 	iter->nnz_idx = 0 ;
 	iter->row_idx = 0 ;
 	iter->nrows = nrows ;
 	iter->p = A->p[0] ;
 	if(iter->sparsity_type == GxB_HYPERSPARSE) {
-		iter->sparse_row_idx = 0 ;
-		iter->h_size = A->nvec ;
+		_MatrixTupleIter_init_hypersparse_fields(iter, 0, A->nvec);
 	}
 
 	return GrB_SUCCESS ;
 }
 
-// Create a new iterator
-GrB_Info GxB_MatrixTupleIter_new
+// finds the start row index in Ah for HYPERSPARSE matrix,
+// returns true if found else false.
+static inline bool _find_minimal_row_in_Ah_greater_or_equal_to_rowIdx
```

Comment:
Why did you decide to mark this function as `inline` ?

Reply:
Because it might be called many time in the future, and it's static, so why not?